microsoft-edge/progressive-web-apps-chromium/how-to/service-workers.md

```diff
@@ -41,63 +73,282 @@ The lifecycle of a Service Worker consists of multiple steps, with each step tri
+    cache.addAll(PRE_CACHED_RESOURCES);
+  }
+
+  event.waitUntil(preCacheResources());
+});
+
+self.addEventListener("fetch", event => {
+  async function navigateOrDisplayOfflinePage() {
+    try {
+      // Try to load the page from the network.
+      const networkResponse = await fetch(event.request);
+      return networkResponse;
+    } catch (error) {
+      // The network call failed, the device is offline.
+      const cache = await caches.open(CACHE_NAME);
+      const cachedResponse = await cache.match("/offline");
+      return cachedResponse;
+    }
+  }
+
+  // We only want to call event.respondWith() if this is a navigation request
+  // for an HTML page.
+  if (event.request.mode === 'navigate') {
+    event.respondWith(navigateOrDisplayOfflinePage());
+  }
+});
+```
+
+
+<!-- ====================================================================== -->
+## Update your Service Worker
+
+#### Install a new Service Worker version
+
+If you make changes to your Service Worker code and deploy the new Service Worker file to your web server, your users' devices will start using the new Service Worker gradually.
+
+Every time a user navigates to one of your app's pages, the browser that's running the app checks if a new version of the Service Worker is available on the server. The browser detects new versions by comparing the contents between the existing Service Worker and the new Service Worker. When a change is detected, the new Service Worker is installed (its `install` event is triggered) and then waits for the existing Service Worker to stop being used on the device.
+
+In practice, this means that there can be two Service Workers running at the same time, but only one intercepts the app's network requests. When the app is closed, the existing Service Worker stops being used. The next time the app is opened, the new Service Worker is activated. The `activate` event is triggered, and the new Service Worker starts intercepting `fetch` events.
+
+You can forcefully activate the new Service Worker as soon as it's installed by using `self.skipWaiting()` in your Service Worker's `install` event handler.
+
+To learn more about how Service Workers update, see [Updating the service worker](https://web.dev/service-worker-lifecycle#updates) on web.dev.
+
+#### Update your cached static files
+
+When pre-caching static resources such as CSS stylesheet files, as described in [Pre-cache resources](#pre-cache-resources), your app only uses the cached versions of the files and doesn't try to download new versions.
+
+To make sure users get the latest changes to the static resources used by your app, use a cache busting naming convention and update your Service Worker code.
+
+Cache-busting means that each static file is named according to its version. This can be achieved in various ways by usually involves using a build tool that reads the content of a file and generates a unique ID based on the content which can be used to name the file. 
```

Comment:
```suggestion
_Cache-busting_ means that each static file is named according to its version. This can be achieved in various ways by usually involves using a build tool that reads the content of a file and generates a unique ID based on the content which can be used to name the file. 
```